pkg/proof/proof_test.go

```diff
@@ -221,3 +221,35 @@ func TestNewShareInclusionProof(t *testing.T) {
 		})
 	}
 }
+
+// TestAllSharesInclusionProof creates proof for all the shares in a block.
+// Since we can't prove multiple namespaces at the moment, the block will contain all the shares with the same namespace.
+func TestAllSharesInclusionProof(t *testing.T) {
+	txs := testfactory.GenerateRandomTxs(64, 500)
+
+	dataSquare, err := square.Construct(txs.ToSliceOfBytes(), appconsts.LatestVersion, appconsts.SquareSizeUpperBound(appconsts.LatestVersion))
+	if err != nil {
+		panic(err)
+	}
+
+	// erasure the data square which we use to create the data root.
+	eds, err := da.ExtendShares(shares.ToBytes(dataSquare))
+	require.NoError(t, err)
+
+	// create the new data root by creating the data availability header (merkle
+	// roots of each row and col of the erasure data).
+	dah, err := da.NewDataAvailabilityHeader(eds)
+	require.NoError(t, err)
+	dataRoot := dah.Hash()
+
+	actualNID, err := proof.ParseNamespace(dataSquare, 0, 64)
```

Comment:
[blocking][question] should this test assert that the len of the data square is 64 shares? When I add

```
	assert.Equal(t, 64, len(dataSquare))
```

I observe an error that makes me think the data square actually has 256 shares in which case this test doesn't actually verify the last share share. Also I confirmed that this test passes if I revert the code change back to

```go
	if endShare >= len(rawShares) {
```

Reply:
good catch [b747251](https://github.com/celestiaorg/celestia-app/pull/2208/commits/b747251305c05c3955d6e34930500612004be0ee)

Now the created square has 256 tx shares and it's proving them

Reply:
What I am not sure of is this:
```
square.Construct(txs.ToSliceOfBytes(), appconsts.LatestVersion, 128)
```
I give the `128` square size but the output is 256 shares... still investigating why is that

Reply:
128 is the max square size, so what is likely happening is we're getting a 16x16 square as there are only enough txs for that

Reply:
aaaaah, makes sense, thanks :+1:

Reply:
resolving